src/Regridder.jl

```diff
@@ -633,5 +633,74 @@ function cgll2latlonz(field; DIR = "cgll2latlonz_dir", nlat = 360, nlon = 720, c
     return new_data, coords
 end
 
+"""
+    truncate_dataset(datafile, name, datapath_trunc, date0, time_start, time_end, comms_ctx)
+
+Truncates given data set, and constructs a new dataset containing only the dates that are used in the simulation
+"""
+function truncate_dataset(
+    datafile,
+    name,
+    datapath_trunc,
+    date0,
+    time_start,
+    time_end,
+    comms_ctx::ClimaComms.AbstractCommsContext,
+)
+    date_start = date0 + Dates.Second(time_start)
+    date_end = date0 + Dates.Second(time_start + time_end)
+
+    file_name = replace(string(name, "_truncated_data_", string(date_start), string(date_end), ".nc"), r":" => "")
+    datafile_truncated = joinpath(datapath_trunc, file_name)
+
+    if ClimaComms.iamroot(comms_ctx)
+        ds = NCDataset(datafile, "r")
+        dates = ds["time"][:]
+
+        # if the simulation start date is before our first date in the dataset
```

Comment:
Could you please split these if else statements (L661 - L694) into a separate function (e.g. `find_bounding_dates`)? This way this can be used for different contexts, and it'll be easier to test.